packages/core/protocols/src/proto/dxos/edge/messenger.proto

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+//syntax = "proto3";
+
+import "google/protobuf/any.proto";
+
+//
+// The messenger protocol supports routing messages between peers.
+//
+package dxos.edge.messenger;
+
+// TODO(burdon): Opaque peer address? To decouple; e.g., for connection muxing.
+
+// TODO(burdon): Design doc.
+//  - How are peers addressed on the network?
+//  - How is a peer identified when you send a point-to-point message to it?
+//  - Do peers without a HALO (that are joining existing identity) have the indentity key missing? How are they identified.
+//  - Do point-to-point messages need to go through a swarm, or can routers communicate directly?
+
+//
+// Network peer.
+//
+// TODO(burdon): Distinguish between point-to-point and broadcast messaging.
+// TODO(burdon): Rename PeerInfo?
+message Peer {
+  // TODO(burdon): Rename topic?
+  optional string swarmKey = 1;
+  optional string identityKey = 2;
+  required string peerKey = 3;
+}
+
+//
+// Message envelope.
+//
+message Message {
+  optional string timestamp = 1;
+
+  // Routing.
+  optional Peer source = 10;
+  repeated Peer target = 11;
+
+  // TODO(burdon): Routing info.
+  optional string service_id = 20;
```

Comment:
define service ID constants

Reply:
can't use enums here cause service_ids are dynamic, defined prefixes in a code file instead